CMakeLists.txt

```diff
@@ -66,6 +66,7 @@ elseif(APPLE)
     add_definitions(-DVK_USE_PLATFORM_MACOS_MVK)
 elseif(UNIX)
     add_definitions(-DVK_USE_PLATFORM_XCB_KHR)
+    add_definitions(-DVK_USE_PLATFORM_WAYLAND_KHR)
```

Comment:
Should we not be testing whether Wayland and Xcb are available before adding definitions to suggest they are support?

Reply:
Qt applications can run on different platforms without having to recompile, thanks to QPA.
VK_USE_PLATFORM_WAYLAND_KHR just includes the vulkan header with the wayland surface extension name
```
#ifdef VK_USE_PLATFORM_WAYLAND_KHR
#include "vulkan_wayland.h"
#endif
```
We can find out the platform only in runtime, so we need to compile the application to work with both platforms.

P.s.
I can't imagine how we can define a windowing system from a CMake.